sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/policy/InstrumentationPolicy.java

```diff
@@ -107,10 +114,18 @@ private Context startSpan(HttpPipelineCallContext azContext) {
         return span;
     }
 
+    private static int getPort(URL url) {
+        int port = url.getPort();
+        if (port == -1) {
+            port = url.getDefaultPort();
+        }
+        return port;
+    }
+
     private void addPostSamplingAttributes(Context span, HttpRequest request) {
-        String userAgent = request.getHeaders().getValue(HttpHeaderName.USER_AGENT);
-        if (!CoreUtils.isNullOrEmpty(userAgent)) {
-            tracer.setAttribute(HTTP_USER_AGENT, userAgent, span);
+        Object rawRetryCount = span.getData(RETRY_COUNT_CONTEXT).orElse(null);
```

Comment:
Comment for myself: Investigate adding metadata to `HttpRequest` where well-known values, such as retry count, can be propagated without the "magic" of Context.

Reply:
common to both of us: try count and maybe other things like user-agent or x-ms-client-request-id variations are only necessary as logging scope - perhaps we don't need to know what's in it but can work with it as it was a bag of key-value-pairs